File: Bidding/Models/ViewModels/Users/Details/UserAdvancedDetailsResponseModel.cs
using Bidding.Models.ViewModels.Users.Shared;
using System;
using System.Collections.Generic;

namespace Bidding.Models.ViewModels.Users.Details
{
    public class UserAdvancedDetailsResponseModel : UserBasicDetailsResponseModel
    {
        /// <summary>
        /// Selected user assigned role id.
        /// </summary>
        public int RoleId { get; set; }

        /// <summary>
        /// All possible roles in the system.
        /// </summary>
        public List<RoleItemModel> Roles { get; set; }
        public DateTime? SubscriptionTill { get; set; }
    }
}

File: Bidding/ClientApp/src/app/users/models/details/user-advanced-details-response.model.ts
import { RoleItemModel } from "../shared/role-item.model";
import { UserBasicDetailsResponseModel } from "./user-basic-details-response.model";

export interface UserAdvancedDetailsResponseModel
  extends UserBasicDetailsResponseModel {
  roleId: number;
  roles: RoleItemModel[];
  subscriptionTill;
}
